test-adapter/src/main/scala/org/scalajs/testing/adapter/TestAdapter.scala

```diff
@@ -47,6 +47,26 @@ final class TestAdapter(jsEnv: JSEnv, input: Seq[Input], config: TestAdapter.Con
   private implicit val executionContext: ExecutionContext =
     ExecutionContext.fromExecutor(ExecutionContext.global, reportFailure)
 
+  /** Accessor to the deprecated method `jl.Thread.getId()`.
+   *
+   *  Since JDK 19, Thread.getId() is deprecated in favor of Thread.threadId().
+   *  The only reason is that getId() was not marked `final`, and hence there
+   *  was no guarantee that subclasses wouldn't override it with something that
+   *  is not the thread's ID.
+   *
+   *  We cannot directly use Thread.threadId() since it was only added in JDK 19.
+   *  Since we probably don't need to care about the potential "threat", we do
+   *  the override-with-deprecated dance to silence the warning.
+   *
+   *  Reminder: we cannot use `@nowarn` since it was only introduced in
+   *  Scala 2.12.13/2.13.2.
```

Comment:
Unrelated to this PR, but this is a smell in our build: `test-adapater` is a normal Scala package, so IIUC, there is really no need we build it with anything but a single Scala patch version (for each Scala minor version).

Reply:
True. I think we tried that in the past to some extent, but did not succeed. We could give it another go.